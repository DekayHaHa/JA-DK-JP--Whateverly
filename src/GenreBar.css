.genre-check {
  box-sizing: border-box;
  padding: 30px;
  position: absolute;
  bottom: 100px;
  width: 90%;
  display: flex;
  flex-wrap: wrap;
  background: hsla(233, 10%, 14%, 0.9);
  box-shadow: 0px 4px 18px #ce63ff;
  border: solid #ce63ff;
  border-width: 0;
  border-radius: 50px 50px 50px 50px;
} 

.hide {
  display: none;
}

.genre-container {
  width: 230px;
}

.genre-container label {
  display: flex;
  align-items: center;
  margin: 10px 0 10px 0;
}

.genre-container:hover .faux-box {
  background: #a24457;
}

.genre-container:hover .genre-title {
  color: #a24457;
}

.genre-container:active .faux-box {
  opacity: 0.5;
}

.genre-title {
  margin: 0 50px 0 10px;
  color: #fff;
  font-weight: 700;
  font-size: 20px;
  transition: all .2s ease;
}

.faux-box {
  background: #ce63ff;
  width: 30px;
  height: 30px;
  border-radius: 5px;
  transition: all .2s ease;
  position: relative;
}

input[type=checkbox] {
  display: none;
}

input[type=checkbox]:checked ~ div.faux-box:after {
  position: absolute;
  left: 8px;
  top: 2px;
  content: "";
  width: 8px;
  height: 16px;
  border: solid #fff;
  border-width: 0 5px 5px 0;
  transform: rotate(45deg);
}