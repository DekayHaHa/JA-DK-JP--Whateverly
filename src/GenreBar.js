import React, { Component } from 'react';
import CheckBox from './CheckBox.js';
import './GenreBar.css'


export default class GenreBar extends Component {
    constructor(props) {
        super(props);
        this.state = {
            genresToFilter: [], 
            showGenres: false
        }
    }
    updateGenres = () => {
        let checked = [];
        const checkboxes = document.querySelectorAll('.genre-check-box')
        checkboxes.forEach((elem) => {
            if (elem.checked === true) {
                checked.push(elem.id)
            }
        })
        this.setState({genresToFilter: checked}, () => {
            this.props.updateState({filteredGames: this.filterGames()})
        });
    }
    filterGames = () => {
        let genres = this.state.genresToFilter;
        return this.props.games.filter(val => {
            return genres.every(elem => val.genres.includes(elem))
        })
    }
    toggleGenres = () => {
        if (this.state.showGenres) {
            this.setState({showGenres: false})
        } else {
            this.setState({ showGenres: true });   
        }
    }
    render() {
        let hideGenres = "";
        if (this.state.showGenres) {
            hideGenres = "genre-check";
        } else {
            hideGenres = "genre-check hide"
        }
        return (
            <div className="genre-bar search-section">
                <form onChange={this.updateGenres} className={hideGenres}>
                {
                    this.props.genres.map((val, i) => {
                        return <CheckBox {...this.props}
                                         name={val}
                                         class="genre-check-box"
                                         key={i}/>
                    })
                }
                </form>
                <button onClick={this.toggleGenres} className="button">GENRES</button>
            </div>
        )
    }
}