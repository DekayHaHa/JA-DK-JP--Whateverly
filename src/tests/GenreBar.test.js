import React from 'react';
import ReactDOM from 'react-dom';
import GenreBar from '../GenreBar';
import { shallow } from 'enzyme';

const mockGenres = ['Action', 'Adventure', 'Fantasy']
const mockFilters = ['Action', 'Adventure']

describe('GenreBar', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <GenreBar 
        genres={mockGenres}
        genresToFilter={mockFilters}
      />
    );
  });
  it('should match snapshop', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  it('should have default state', () => {
    expect(wrapper.state()).toEqual({
      showGenres: false
    });
  })
})