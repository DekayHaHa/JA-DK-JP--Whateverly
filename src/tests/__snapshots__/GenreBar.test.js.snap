// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreBar should match snapshop 1`] = `
<div
  className="genre-bar search-section"
>
  <form
    className="genre-check hide"
    onChange={[Function]}
  >
    <Footer
      class="genre-check-box"
      key="0"
      name="Action"
    />
    <Footer
      class="genre-check-box"
      key="1"
      name="Adventure"
    />
    <Footer
      class="genre-check-box"
      key="2"
      name="Fantasy"
    />
  </form>
  <button
    className="button"
    onClick={[Function]}
  >
    GENRES
  </button>
</div>
`;
